### PR TITLE
nfadeeva-assignment_1

### PR DESCRIPTION
**Pull request by nfadeeva**
Add solutions for assignment_1

### CODE REVIEW

File: nfadeeva/assignment1/Assignment_1_Q1.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    # delete from the count_s1 every word that is anagram of the word in count_s2
+    for word in count_s2:
+        if word not in count_s1:
+            return False
+        else:
+            count_s1.remove(word)
+
+    # if all words from the s2 is anagrams of one of the words in s1 return True
+    return len(count_s1) == 0
```
Comment (ariava) [inline nfadeeva/assignment1/Assignment_1_Q1.py]: Please respect the convention of having one blank line at the end of the file.

File: nfadeeva/assignment1/Assignment_1_Q1.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import string
+from collections import Counter
+import re
+
+
+def is_anagram(s1, s2, case_sensitive=False, punctuation=string.punctuation):
+    """
```
Comment (ariava) [inline nfadeeva/assignment1/Assignment_1_Q1.py:7]: Awesome docstring! :D

File: nfadeeva/assignment1/Assignment_1_Q2.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    def push(self, value):
+        """
+        Push value to the head of the list
+        """
+
+        new_node = Node(value)
+        new_node.next = self.head
+        self.head = new_node
+
+    def find_kth_last_element(self, k):
```
Comment (ariava) [inline nfadeeva/assignment1/Assignment_1_Q2.py]: nit: as it's a method of the LinkedList I would just call it "kth_last_element(...)".

File: nfadeeva/assignment1/Assignment_1_Q1.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    if not case_sensitive:
+        s1 = s1.lower()
+        s2 = s2.lower()
+
+    # remove all whitespaces
+    s1 = re.sub(r"\s+", " ", s1)
+    s2 = re.sub(r"\s+", " ", s2)
+
+    # count characters for every word in both sentences
+    count_s1 = [Counter(x) for x in s1.strip().split(' ')]
```
Comment (ariava) [inline nfadeeva/assignment1/Assignment_1_Q1.py:29]: You are using Counter, super!

**Review by ariava — CHANGES_REQUESTED** [review]
Awesome, I really like your attention to comments! Please have a look at my review.

File: nfadeeva/assignment1/Assignment_1_Q1.py (hunk 1 above)
Comment (nfadeeva) [reply nfadeeva/assignment1/Assignment_1_Q1.py]: Fix it and also rewrite the last string in more "pythonic" way:)

File: nfadeeva/assignment1/Assignment_1_Q2.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    def push(self, value):
+        """
+        Push value to the head of the list
+        """
+
+        new_node = Node(value)
+        new_node.next = self.head
+        self.head = new_node
+
+    def kth_last_element(self, k):
```
Comment (ariava) [inline nfadeeva/assignment1/Assignment_1_Q2.py:20]: Can you add a few examples to show that the function really works? :)

File: nfadeeva/assignment1/Assignment_1_Q1.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import string
+from collections import Counter
+import re
+
+
+def is_anagram(s1, s2, case_sensitive=False, punctuation=string.punctuation):
```
Comment (ariava) [inline nfadeeva/assignment1/Assignment_1_Q1.py:6]: Can you add a few examples to show that the function really works? :)

**Review by ariava — CHANGES_REQUESTED** [review]

File: nfadeeva/assignment1/Assignment_1_Q2.py (hunk 5 above)
Comment (nfadeeva) [reply nfadeeva/assignment1/Assignment_1_Q2.py:20]: Examples of usage were added:)

**Review by ariava — APPROVED** [review]
Looks great to me!

**Pull request merged**